Assignment2/Document.cpp

```diff
@@ -0,0 +1,79 @@
+
+
+#include "document.h"
+
+#include<fstream>
+#include<iostream>
+#include<string>
+#include<vector>
+#include<algorithm>
+#include<map>
+#include<iterator>
+#include<iomanip>
+#include <math.h> 
+
+typedef  map<string, map<string, unsigned int >> nestedMap;
+using namespace std;
+
+//defult constructor that creates an empty document
+Document::Document()
+{
+	ofstream d("Document.txt");
+	d << "";
+	d.close();
+}
+
+//constructor that accepts a file name and reads the files into the document object
+Document::Document(string fname)
+
+	:documentName(fname)
+{
+	ifstream txtfile(documentName);
+
+	while (txtfile >> word)

```

Comment:
Tokenizer should be used here before pushing the words into the vector (or map like I suggested)

Reply:
you want to store it in a map like map[word1]=0; map[word2]=0, map[word3]=0?? etc

Reply:
yeah or in this case just map[word]++ like we did in assignment #1